controller/volume_controller.go

```diff
@@ -2999,6 +3000,92 @@ func (vc *VolumeController) ResolveRefAndEnqueue(namespace string, ref *metav1.O
 	vc.enqueueVolume(volume)
 }
 
+// createRecurringJobNotExistAndLabel add recurring jobs and group back to a restoring volume and create recurring jobs if not exist.
+func createRecurringJobNotExistAndLabel(log *logrus.Entry, ds *datastore.DataStore, v *longhorn.Volume, jobName string, job *longhorn.RecordVolumeRecurringJob) error {
+	if existJob, err := ds.GetRecurringJob(jobName); err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.WithError(err).Warnf("Failed to get the recurring job %v", jobName)
+			return err
+		}
+		// create the recurring job if not exist
+		log.Debugf("Create the recurring job %v when restoring recurring jobs", jobName)
+		if _, err := ds.CreateRecurringJob(&longhorn.RecurringJob{ObjectMeta: metav1.ObjectMeta{Name: jobName}, Spec: job.JobSpec}); err != nil {
+			log.WithError(err).Warnf("Failed to create a recurring job %v", jobName)
+			return err
+		}
+	} else {
+		// create the recurring job if job name is used and configuration is different.
+		if !reflect.DeepEqual(existJob.Spec, job.JobSpec) {
+			newJobName := util.RandomID()
+			if _, err := ds.GetRecurringJob(newJobName); err != nil {
+				if apierrors.IsNotFound(err) {
+					newJobSpec := job.JobSpec.DeepCopy()
+					newJobSpec.Name = newJobName
+					log.Debugf("Re-create the recurring job %v instead of a existing job: %v when restoring recurring jobs", newJobName, jobName)
+					if _, err := ds.CreateRecurringJob(&longhorn.RecurringJob{ObjectMeta: metav1.ObjectMeta{Name: newJobName}, Spec: *newJobSpec}); err != nil {
+						log.WithError(err).Warnf("Failed to create a recurring job %v", jobName)
+						return err
+					}
+					jobName = newJobName
+				}
+			}
+		}
+	}
+
+	// set Volume recurring jobs/groups
+	if job.FromJob {
+		labelKey := types.GetRecurringJobLabelKey(types.LonghornLabelRecurringJob, jobName)
+		// disabled recurring jobs/group and it would be enabled after the backup restoration is completed.
+		v.Labels[labelKey] = types.LonghornLabelValueEnabled
+	}
+	for _, groupName := range job.FromGroup {
+		groupLabelKey := types.GetRecurringJobLabelKey(types.LonghornLabelRecurringJobGroup, groupName)
+		// disabled recurring jobs/group and it would be enabled after the backup restoration is completed.
+		v.Labels[groupLabelKey] = types.LonghornLabelValueEnabled
+	}
```

Comment:
Sorry about the misleading of `// disabled recurring jobs/group and it would be enabled after the backup restoration is ...`, now it will be enabled at this moment.

Reply:
Sorry... I am still confused about why we need this label... If enabling the recurring job restore should be already controlled by `v.Spec.RestoreVolumeRecurringJob` or the global setting...

Reply:
`v.Spec.RestoreVolumeRecurringJob` is used to check if it needed to restore recurring jobs from remote backup target or not. And `labelKey/groupLabelKey` label is used to filter volumes that is set to the job/group when recurring job is start to work.

Reply:
Then when/where is this label used to filter volumes?

Reply:
https://github.com/longhorn/longhorn-manager/blob/master/app/recurring_job.go#L136-L149
when recurring job (pod created by the cronjob) starts to do snapshot or backup.